dapp/src/components/Header/ConnectWallet.tsx

```diff
@@ -63,7 +70,47 @@ export default function ConnectWallet() {
     )
   }
 
-  return (
+  return isMobile ? (
+    <Menu>
+      {({ isOpen }) => (
+        <>
+          <MenuButton
+            as={Button}
+            variant="card"
+            leftIcon={<Icon as={BitcoinIcon} boxSize={6} color="brand.400" />}
+            rightIcon={isOpen ? <IconChevronUp /> : <IconChevronDown />}
+          >
+            <TextMd color="brand.400">{truncateAddress(address)}</TextMd>
+          </MenuButton>
+          <MenuList bg="gold.200">
+            <MenuItem
+              {...styles}
+              icon={<Icon as={IconCopy} boxSize={5} />}
+              onClick={onCopy}
+            >
+              {hasCopied ? "Address copied" : "Copy Address"}
```

Comment:
It doesn't quite work well. "Address copied"  shows up when we open the menu again.

https://github.com/user-attachments/assets/9d952999-a10d-4286-a8fa-51b869346d86

Reply:
Yeah. Not sure how to handle it.  You see `Address copied` because the `useClipboard` hook has a default timeout set to `1500 ms`.  We can get rid of the conditional rendering or leave it as is for now because the mobile UI will change soon. cc @nkuba

Reply:
68c5c99. I keep the drop-down open to display the `Address copied` message. The drop-down will close once users select other options.

Reply:
I agree, the the mobile UI will change soon. So for now let's leave it as it is.